task.js

```diff
@@ -0,0 +1,193 @@
+// you can see an example of interaction with the order at the end of the file
+
+class Item {
+    constructor(price, calories) {
+        this.price = price;
+        this.calories = calories;
+    }
+
+    calculatePrice() {
+        throw new Error("abstract method");
+    }
+
+    calculateCalories() {
+        throw new Error("abstract method");
+    }
+}
+
+class HamburgerStuffing {
+    constructor(price, calories) {
+        this.price = price;
+        this.calories = calories;
+    }
+
+    getPrice() {
+        return this.price;
+    }
+
+    getCalories() {
+        return this.calories;
+    }
+}
+
+const cheese = new HamburgerStuffing(10, 20);
+const salad = new HamburgerStuffing(20, 5);
+const potato = new HamburgerStuffing(15, 10);
+
+class Hamburger extends Item {
+    constructor(price, calories, stuffing) {
```

Comment:
In general, your version of inheritance will work, but I am not sure that it's fully correct.
I do not agree that small hamburger, big hamburger and so on are separated classes - it must be only instances of the class `Hamburger`, `Salad` and so on.

You shouldn't create new class in case **you only change parameters in constructor**.